### PR TITLE
Use DB for Beef storage

### LINKED ISSUE #5: Add sorting and searching of Beef list

### PR DESCRIPTION
**Pull request by SebastianJ21**
Closes #5

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -86,11 +84,14 @@ export default function Home() {
           <Stack p={4} gap={2}>
             <Stack direction="row" justifyContent="space-between">
               <Typography variant="h3">Beef List 🥩📝</Typography>
-              <Link href="/beef/new" style={{ textDecoration: "none" }}>
-                <Button variant="contained" color="secondary">
-                  Start beef
-                </Button>
-              </Link>
+              <Stack direction="row" gap={2}>
+                <BeefSortDropdown sortOption={sortOption} setSortOption={setSortOption} />
```
Comment (memoriesadrift) [inline src/app/page.tsx:98]: Add a title to this so it's clear what it does

File: src/server/actions/beef/arbiterChangeStatus.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { activeChain } from "@/utils/chain"
+
+export const refreshArbiterStatusesAction = async (beefAddress: Address) => {
+  const beef = await db.query.Beefs.findFirst({
+    where: (Beefs, { eq, and }) => and(eq(Beefs.chainId, activeChain.id), eq(Beefs.address, beefAddress)),
+    columns: { arbiters: true },
+  }).execute()
+
+  if (beef === undefined) {
+    throw new Error(`Beef with address ${beefAddress} does not exist`)
```
Comment (memoriesadrift) [inline src/server/actions/beef/arbiterChangeStatus.ts:17]: Throw or return `true` seems like a bit of a weird pattern to me. Shouldn't we return a unified type? `bool`, `Ok | Error<string>` or just `void`?

File: src/server/actions/beef/refreshBeefState.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+export const refreshBeefStateAction = async (beefAddress: Address) => {
+  const { cooking, beefGone } = await getBeefContractInfo(beefAddress)
+
+  await db
+    .update(Beefs)
+    .set({ isCooking: cooking, beefGone })
+    .where(and(eq(Beefs.chainId, activeChain.id), eq(Beefs.address, beefAddress)))
+    .execute()
+
+  return true
```
Comment (memoriesadrift) [inline src/server/actions/beef/refreshBeefState.ts]: Similar comment -- why are we always returning true? A function that always succeeds should be `void` in my book.

File: src/server/actions/getGeneratedSmartAccount.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    console.error("Auth token not found")
+    return undefined
+  }
+
+  const claims = await privy.verifyAuthToken(authToken)
+  const user = await privy.getUser(claims.userId)
+
+  const walletAddress = user.linkedAccounts.find(isWalletWithMetadata)?.address as Address
+
+  if (walletAddress === undefined) {
```
Comment (memoriesadrift) [inline src/server/actions/getGeneratedSmartAccount.ts]: casting to `Address` and then immediatelly using it as if it were of type `Address | undefined` seems like a mistake

File: src/server/actions/getGeneratedSmartAccount.ts [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    console.error(`User ${user.id} does not have an embedded wallet`)
+    return undefined
+  }
+
+  const xHandle = user.twitter?.username ?? undefined
```
Comment (memoriesadrift) [inline src/server/actions/getGeneratedSmartAccount.ts:39]: Can't we drop the `?? undefined`? or is it to go from `null` -> `undefined`

**Review by memoriesadrift — CHANGES_REQUESTED** [review]

File: src/app/page.tsx (hunk 1 above)
Comment (SebastianJ21) [reply src/app/page.tsx:98]: What do you mean? A title in the UI?

File: src/server/actions/getGeneratedSmartAccount.ts (hunk 4 above)
Comment (SebastianJ21) [reply src/server/actions/getGeneratedSmartAccount.ts]: Yeah, I will split the `undefined` check and walletAddress cast

File: src/server/actions/getGeneratedSmartAccount.ts (hunk 5 above)
Comment (SebastianJ21) [reply src/server/actions/getGeneratedSmartAccount.ts:39]: Yeah, it's for the null

File: src/server/actions/beef/arbiterChangeStatus.ts (hunk 2 above)
Comment (SebastianJ21) [reply src/server/actions/beef/arbiterChangeStatus.ts:17]: Agreed, I put the `true` as a placeholder. `void` makes way more sense for now

File: src/app/page.tsx (hunk 1 above)
Comment (memoriesadrift) [reply src/app/page.tsx:98]: Yes, at least a small one like in mui outlined text area https://mui.com/material-ui/react-text-field/#inputs

**Review by memoriesadrift — APPROVED** [review]
Awesome, thanks for implementing the changes quickly!

**Comment by memoriesadrift** [comment]
Should the base of this PR be changed to `main`?

**Pull request merged**